TestAppExample/keychainclass.cpp

```diff
@@ -0,0 +1,61 @@
+#include <QDebug>
+
+#include "keychainclass.h"
+
+KeyChainClass::KeyChainClass(QObject* parent) :
+    QObject(parent),
+    m_readCredentialJob(QLatin1String("keychain.example.project.app")),
+    m_writeCredentialJob(QLatin1String("keychain.example.project.app")),
+    m_deleteCredentialJob(QLatin1String("keychain.example.project.app"))
+{
+    m_readCredentialJob.setAutoDelete(false);
+    m_writeCredentialJob.setAutoDelete(false);
+    m_deleteCredentialJob.setAutoDelete(false);
+}
+
+void KeyChainClass::readKey(const QString &key)
+{
+    m_readCredentialJob.setKey(key);
+
+    QObject::connect(&m_readCredentialJob, &QKeychain::ReadPasswordJob::finished, [=](){
+        if (m_readCredentialJob.error()) {
+            emit error(QString("Write key failed: %1").arg(qPrintable(m_readCredentialJob.errorString())));
```

Comment:
"Write key failed" should read "Read key failed". Also use QStringLiteral() or maybe better, tr() (although nobody will ever translate this test app, I guess)